src/Sentry/Platforms/Native/UnmanagedHttpContent.cs

```diff
@@ -0,0 +1,63 @@
+using Sentry.Extensibility;
+
+namespace Sentry.Native;
+
+internal class UnmanagedHttpContent : SerializableHttpContent
+{
+    private readonly IntPtr _content;
+    private readonly int _length = 0;
+    private readonly IDiagnosticLogger? _logger;
+
+    public UnmanagedHttpContent(IntPtr content, int length, IDiagnosticLogger? logger)
+    {
+        _content = content;
+        _length = length;
+        _logger = logger;
+    }
+
+    protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context)
+    {
+        try
+        {
+            unsafe
+            {
+                using var unmanagedStream = new UnmanagedMemoryStream((byte*)_content.ToPointer(), _length);
+                return unmanagedStream.CopyToAsync(stream);
+            }
+        }
+        catch (Exception e)
+        {
+            _logger?.LogError(e, "Failed to serialize unmanaged content into the network stream");
+            throw;
+        }
+    }
+
+    protected override void SerializeToStream(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+    {
+        try
+        {
+            unsafe
+            {
+                using var unmanagedStream = new UnmanagedMemoryStream((byte*)_content.ToPointer(), _length);
+                unmanagedStream.CopyTo(stream);
+            }
+        }
+        catch (Exception e)
+        {
+            _logger?.LogError(e, "Failed to serialize unmanaged content into the network stream");
+            throw;
+        }
+    }
+
+    protected override bool TryComputeLength(out long length)
+    {
+        length = _length;
+        return false;
```

Comment:
question: return `true`

I would have expected to return `true` here, to indicate there we were able to compute the length.
Or are there scenarios, where the `int length` passed in the constructor is not valid (like `0` or `-1` or similar)?

Reply:
Thanks, good catch! My bad while copying `EnvelopeHttpContent` :)